### DISCUSSION TITLE
Changing strings from lowercase to uppercase

### DISCUSSION THREAD

**Discussion by awa5114** [lvjr/functional#6] [Q&A]
Is there a way using `functional` to change a string or character from uppercase to lowercase and vice versa? E.g.:

- `hello` to `Hello`
- `hello` to `HELLO`
- `hello` to `heLLo`

I've been looking through `functional.pdf` but I can't find anything about that...

**Comment by lvjr** [comment]
I have added several functions  for changing text or character between uppercase and lowercase. You can download the latest [package file](https://github.com/lvjr/functional/raw/main/functional.sty) and test whether there are other problems. 
```latex
\documentclass{article}

\usepackage{functional}

\begin{document}

%% hello => Hello
\textTitlecase {hello}

%% hello => HELLO
\textUppercase {hello}

%% l => L
\charUppercase l

%% hello => heLLo
\begingroup
\charSetLccode {`\l} {`\L}
\textLowercase {hello}
\endgroup

\end{document}
```

![image](https://user-images.githubusercontent.com/4239949/164888093-f258da1c-fc9e-400d-884f-20ec55c7f120.png)

**Comment by lvjr** [comment]
From version 2022G, all function names are changed from UpperCamelCase to loweCamelCase, to avoid potential name conflicts with kernel and other packages. And the above example has been updated.
